internal/db/schema/migrations/postgres/16/01_plugin.up.sql

```diff
@@ -0,0 +1,78 @@
+begin;
+
+  -- We are updating the plugin table here to have a scope id, since
+  -- subtypes of plugin should be scoped.  We use 'global' as the default
+  -- just for the purpose of the migration since the scope cannot be null
+  -- and the one plugin that already exists (pi_system) can be globally
+  -- scoped.  The immediately following statement removes this default.
+  -- We add a name so we can enforce that all plugins have a unique name
+  -- in a specific scope across all plugin subtypes.
+  alter table plugin
+    add column name wt_name,
+    add column scope_id wt_scope_id
+      not null
+      default 'global'
+      constraint plugin_must_be_in_global
+      references iam_scope_global(scope_id)
+        on delete cascade
+        on update cascade;
+
+  alter table plugin
+    alter column scope_id drop default;
+
+  -- Add constraints that enforce names are unique across all plugin subtypes
```

Comment:
Given that for now we have no plans nor design for what non-`global` plugins would look like, can you also add in a constraint that the scope ID is global? We can remove this later if needed.

Reply:
Oop, I missed that in the alter table statement above.